play.pokemonshowdown.com/src/panel-chat.tsx

```diff
@@ -575,26 +574,28 @@ export class ChatRoom extends PSRoom {
 		const userid = toID(name);
 		if (this.args?.format) this.args.format = null;
 
-		if (userid === PS.user.userid) {
-			if (!challenge && !this.challenging) {
-				// this is also used for canceling challenges
-				this.challenged = null;
-			}
-			// we are sending the challenge
-			this.challenging = challenge;
+		// Protocol documentation: https://github.com/smogon/pokemon-showdown-client/pull/1799
+
+		this.teamSent = null;
+		if (!challenge) {
+			// rejected or canceled. maybe also accepted?
+			// (when we reject, we are sender; when we cancel, we are not)
+			this.challenged = null;
+			this.challenging = null;
```

Comment:
Ok. I think this change is bad. It makes it so that when you reject a challenge, you also cancel a challenge, and vice versa, but these are different things. I think the old code didn't have this problem.